### PR TITLE
fix(api): various fixes for overlooked stuff

### PR DESCRIPTION
**Pull request by D4isDAVID**
**Please describe the changes this PR makes and why it should be merged:**

This PR mainly contains corrections for incorrect typings that were overlooked.

In addition this also makes `InteractionsAPI#followUp()` return a result which should always be `Promise<APIMessage>` on success (according to the [Discord docs](https://discord.com/developers/docs/interactions/receiving-and-responding#create-followup-message) on the creation of followup messages: "Functions the same as Execute Webhook, but `wait` is always true.")

**Status and versioning classification:**

- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating

### CODE REVIEW

**Comment by suneettipirneni** [comment]
> In addition this also makes InteractionsAPI#followUp() return a result which should always be Promise<APIMessage> on success

Can you test if this is the case? If I recall correctly, followUp didn't return anything.

**Comment by D4isDAVID** [comment]
> > In addition this also makes InteractionsAPI#followUp() return a result which should always be Promise on success
> 
> Can you test if this is the case? If I recall correctly, followUp didn't return anything.

The following code leaves me with a `message` variable which is an APIMessage object (tried with chat input command, button, string select menu, and modal):

```js
client.on(GatewayDispatchEvents.InteractionCreate, async ({ api, data: interaction }) => {
    await api.interactions.reply(interaction.id, interaction.token, {
        content: 'reply',
    });
    const message = await api.interactions.followUp(interaction.application_id, interaction.token, {
        content: 'followup',
    });
    console.log(message);
});
```

`console.log` result:
Screenshot 2023-05-20 at 2 27 59

Other than this test the Discord docs say that `wait` is always true here meaning that the message should be returned, and also [discord.js v14 docs](https://old.discordjs.dev/#/docs/discord.js/main/class/CommandInteraction?scrollTo=followUp) say `Promise<Message>` too

**Comment by suneettipirneni** [comment]
> > > In addition this also makes InteractionsAPI#followUp() return a result which should always be Promise on success
> > 
> > 
> > Can you test if this is the case? If I recall correctly, followUp didn't return anything.
> 
> The following code leaves me with a `message` variable which is an APIMessage object (tried with chat input command, button, string select menu, and modal):
> 
> ```js
> client.on(GatewayDispatchEvents.InteractionCreate, async ({ api, data: interaction }) => {
>     await api.interactions.reply(interaction.id, interaction.token, {
>         content: 'reply',
>     });
>     const message = await api.interactions.followUp(interaction.application_id, interaction.token, {
>         content: 'followup',
>     });
>     console.log(message);
> });
> ```
> 
> `console.log` result: Screenshot 2023-05-20 at 2 27 59
> 
> Other than this test the Discord docs say that `wait` is always true here meaning that the message should be returned, and also [discord.js v14 docs](https://old.discordjs.dev/#/docs/discord.js/main/class/CommandInteraction?scrollTo=followUp) say `Promise<Message>` too

Apologies I got this endpoint mixed up the the reply endpoint.

**Review by suneettipirneni — APPROVED** [review]

File: packages/core/src/api/interactions.ts [hunk 1]
```diff
@@ -103,7 +103,7 @@ export class InteractionsAPI {
 		body: APIInteractionResponseCallbackData & { files?: RawFile[] },
 		{ signal }: Pick<RequestData, 'signal'> = {},
 	) {
-		await this.webhooks.execute(applicationId, interactionToken, body, { signal });
+		return this.webhooks.execute(applicationId, interactionToken, { ...body, wait: true }, { signal });
```
Comment (Jiralite) [inline packages/core/src/api/interactions.ts:106]: Should `wait` be configurable instead of hardcoded to `true`?

Comment (jaw0r3k) [reply packages/core/src/api/interactions.ts:106]: why not?

**Review by kyranet — APPROVED** [review]

**Review by iCrawl — APPROVED** [review]

**Pull request merged**